### PR TITLE
Handle InputStream when reading MDN text

### LINKED ISSUE #253: MDN Response logging shows as Java object instead of the message string
I'm sending some messages using OpenAS2 and receiving the corresponding MDNs with success but when I look into the stored received MDN files (and the mdn_response field in the db), instead of the text received in the MDN, it shows something like:

...
DISPOSITION: automatic-action/MDN-sent-automatically; processed
Text:
java.io.ByteArrayInputStream@76182f20

and it shows the same in the db mdn_response field:

    java.io.ByteArrayInputStream@76182f20

### PR DESCRIPTION
**Pull request by pkraeutli**
Fixes OpenAS2/OpenAs2App#253

### CODE REVIEW

File: Server/src/main/java/org/openas2/util/AS2Util.java [hunk 1]
```diff
@@ -147,6 +134,16 @@ public static void parseMDN(Message msg, X509Certificate receiver) throws OpenAS
         }
     }
 
+    private static String getMimeBodyPartText(MimeBodyPart part) throws MessagingException, IOException {
+        Object content = part.getContent();
+
+        if (InputStream.class.isAssignableFrom(content.getClass())) {
+            return IOUtils.toString((InputStream) content, StandardCharsets.UTF_8);
```
Comment (pkraeutli) [inline Server/src/main/java/org/openas2/util/AS2Util.java:141]: Not sure how to get the actual charset ...

Comment (uhurusurfa) [reply Server/src/main/java/org/openas2/util/AS2Util.java:141]: After line 119 of parseMDN method add this:
```
                Charset charset = StandardCharsets.UTF_8;
                String charsetFromContentType = reportType.getParameter("charset");
                if (charsetFromContentType != null) {
                	try {
						charset = Charset.forName(charsetFromContentType);
					} catch (Exception e) {
						// Just warn and allow the default to go thru
						logger.warn("Received charset string that cannot be parsed: " + charsetFromContentType + msg.getLogMsgID());
					}
                }
```
Then pass `charset` into the getMimeBodyPartText method.

**Review by uhurusurfa — CHANGES_REQUESTED** [review]
Thanks for the contribution - I have added  comment for determining the character set

**Review by uhurusurfa — APPROVED** [review]
👍 Looks good. Thank you for the contribution to the code base.

**Pull request merged**